api/app.js

```diff
@@ -10,24 +10,58 @@ var indexRouter = require('./routes/index');
 var usersRouter = require('./routes/users');
 var coursesRouter = require('./routes/courses');
 var reviewsRouter = require('./routes/reviews');
+var authRouter = require('./routes/auth-routes');
+
+const cookieSession = require("cookie-session");
+const passport = require('passport');
+require("./google-setup");
 
 var app = express();
 
+app.use(
+  cookieSession({
```

Comment:
how are the cookie session object items defined?

Reply:
Do you mean how is the cookie defined/set? They are handled by passport but we can specify what gets put in the cookie. If you look at line 44, we have a `passport.serializeUser()`, that function allows us to decide what to put in the cookie. In this case, we're passing in the whole user object (from the parameter) into the cookie. We don't get to call the `passport.serializeUser()`. I am not entirely sure but I think the user object(from the parameter) is the user object from the Facebook/Google OAuth response.

Reply:
Interesting!  I always wondered how cookies are handled.  I was also wondering how the items name, keys, maxAge is used?

Reply:
I'm not entirely sure what the name and keys do. but I think the maxAge defines how long the cookie is valid